sympy/solvers/ode.py

```diff
@@ -4190,12 +4183,12 @@ def is_special_case_of(soln1, soln2):
 
     return unique_solns
 
-def _nth_algebraic_is_special_case_of(soln1, soln2, eq, order, var):
+def _is_special_case_of(soln1, soln2, eq, order, var):
     r"""
     True if soln1 is found to be a special case of soln2 wrt some value of the
     constants that appear in soln2. False otherwise.
     """
-    # The solutions returned by nth_algebraic should be given explicitly as in
+    # The solutions returned by dsolve should be given explicitly as in
```

Comment:
The comment above is true for nth algebraic but not for dsolve in general. You'll need to check for implicit solutions now. It might be enough to subtract the sides of each equation.

Reply:
Here I don't understand, what do you want me to do?

Reply:
The full comment there says
```
    # The solutions returned by nth_algebraic should be given explicitly as in
    # Eq(f(x), expr). We will equate the RHSs of the two solutions giving an
    # equation f1(x) = f2(x).
```
It is true that nth_algebraic always returns solutions in this form but some other solvers in dsolve can return implicit solutions like:
```julia
In [32]: dsolve(f(x).diff(x) + 1/(2*f(x) + exp(f(x))))                                                                            
Out[32]: 
     2       f(x)     
x + f (x) + ℯ     = C₁
```
In those cases we can't simply assume that the solutions are given by `sol.rhs`. I think it is enough to use `sol.rhs-sol.lhs` instead though and the rest of the method might still work.

At minimum you should check that the left hand side is `f(x)` before proceeding. It is better to allow some redundant solutions than to incorrectly remove some valid solutions.

I can't think of any examples where we would get multiple solutions in implicit form since the solvers that give implicit solutions only give single solutions when they do so. With factorisable though it wouldn't be hard to come up with examples where two solutions are returned in implicit form that are or are not special cases of one another.

Reply:
> In those cases we can't simply assume that the solutions are given by `sol.rhs`. I think it is enough to use `sol.rhs-sol.lhs` instead though and the rest of the method might still work.


We can send in general `sol.rhs-sol.lhs ` without checking the form of an equation. I tried this and It is working fine. So I will proceed with this.

Reply:
For now, we don't have any problems with the implicit form of the solution because whenever dsolve gives the solution in the form of implicit, then it will be in the form Equation not in the list. And we are passing the only list to the `_remove_redundant_solution`. 

But It may cause the problem in `factorable` method that I want to implement. Because We will return the list of solutions that may contain an implicit and explicit solution in the factorable method.